src/client/connstring.rs

```diff
@@ -0,0 +1,311 @@
+use std::ascii::AsciiExt;
+use std::collections::HashMap;
+use std::vec::Vec;
+
+pub const DEFAULT_PORT: u16 = 27017;
+pub const URI_SCHEME: &'static str = "mongodb://";
+
+/// Encapsulates the hostname and port of a host.
+pub struct Host {
+    pub host_name: String,
+    pub ipc: String,
+    pub port: u16,
+}
+
+impl Host {
+    // Creates a new Host struct.
+    fn new(host_name: String, port: u16) -> Host {
+        Host {
+            host_name: host_name,
+            port: port,
+            ipc: String::new(),
+        }
+    }
+
+    fn new_ipc(ipc: String) -> Host {
```

Comment:
I would think about renaming this function. This makes it sound like it's making a new ipc, when it's making  a new `Host`. There's a small convention for alternate constructors like this: `with_*`. So I'd call it `with_ipc()`.